lib/shrine/plugins/copy.rb

```diff
@@ -21,14 +21,17 @@ module AttachmentMethods
         def initialize(*)
           super
 
-          module_eval <<-RUBY, __FILE__, __LINE__ + 1
-            def initialize_copy(record)
-              super
-              @#{@name}_attacher = nil # reload the attacher
-              #{@name}_attacher.send(:write, nil) # remove original attachment
-              #{@name}_attacher.copy(record.#{@name}_attacher)
-            end
-          RUBY
+          name = attachment_name
+
+          define_method :initialize_copy do |record|
+            super(record)
+            attacher = send("#{name}_attacher", true) # reload the attacher
+            attacher.send(:write, nil) # remove original attachment
+            attacher.copy(record.public_send("#{name}_attacher"))
+          end
+
+          # Fix for JRuby
+          private :initialize_copy
```

Comment:
https://travis-ci.org/shrinerb/shrine/builds/457011731

Reply:
I'm wondering why this wasn't needed for MRI 🤔. Anyway, I'm fine with that, as we'll probably remove that part anyway. I think we can even omit the comment, as I would expect MRI to require this as well.

Reply:
Builds are green for all rubies except jruby. Do you think it's mri issue?

Reply:
It's not, I believe you that it only happens with JRuby, the build clearly shows only JRuby. I just said I would like to have this line regardless, because I wouldn't expect this to work on MRI either (even though it does). Actually, the current comment is fine, no changes needed.